Automattic/jetpack
# jetpack: Use jest for client tests

#### Changes proposed in this Pull Request:

This conversion was remarkably straightforward. Of note is that we now
have separate jest config files for test-extensions and test-client.

Also, since I'm touching the test files anyway, I enabled linting for
them and configured the jest lint rule.

#### Jetpack product discussion


Working towards p4TIVU-a8r-p2#comment-11500

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Check that the right tests are still being run.
* Check that coverage is still being generated too.
